monix-tail/shared/src/main/scala/monix/tail/Iterant.scala

```diff
@@ -1392,6 +1392,24 @@ sealed abstract class Iterant[F[_], A] extends Product with Serializable {
   final def takeWhile(p: A => Boolean)(implicit F: Sync[F]): Iterant[F, A] =
     IterantTakeWhile(self, p)(F)
 
+  /** Takes longest prefix of elements zipped with their indices that satisfy the given predicate
+    * and returns a new iterant that emits those elements.
+    *
+    * Example: {{{
+    *   // Yields 1, 2
+    *   Iterant[Task].of(1, 2, 3, 4, 5, 6).takeWhile((_, idx) => idx != 2)
```

Comment:
An, just one very small issue with this sample, you got the name wrong: `takeWhile` => `takeWhileWithIndex`

😀

Reply:
@alexandru Thanks for spotting this one, Alexandru! I was so obsessed to come up with a good example that I missed it. Fixed.